core/server/master/src/main/java/alluxio/master/file/mdsync/DefaultSyncProcess.java

```diff
@@ -0,0 +1,997 @@
+    } else {
+      syncState.mContext.reportSyncOperationSuccess(SyncOperation.NOOP);
+    }
+    checkShouldSetDescendantsLoaded(currentInode.getInode(), syncState);
+    return new SingleInodeSyncResult(true, true, false);
+  }
+
+  private void handleConcurrentModification(
+      SyncProcessContext context, String path, boolean isRoot, Exception e)
+      throws FileAlreadyExistsException, FileDoesNotExistException {
+    String loggingMessage = "Sync metadata failed on [{}] due to concurrent modification.";
+    if (!isRoot && context.isConcurrentModificationAllowed()) {
+      context.reportSyncOperationSuccess(SyncOperation.SKIPPED_DUE_TO_CONCURRENT_MODIFICATION);
+      LOG.info(loggingMessage, path, e);
+    } else {
+      context.reportSyncFailReason(SyncFailReason.PROCESSING_CONCURRENT_UPDATE_DURING_SYNC, e);
+      LOG.error(loggingMessage, path, e);
+      if (e instanceof FileAlreadyExistsException) {
+        throw (FileAlreadyExistsException) e;
+      }
+      if (e instanceof FileDoesNotExistException) {
+        throw (FileDoesNotExistException) e;
+      }
+      throw new RuntimeException(e);
+    }
+  }
+
+  private Pair<Integer, Integer> deletePath(
+      SyncProcessContext context, LockedInodePath lockedInodePath, boolean skipNonPersisted)
+      throws FileDoesNotExistException, DirectoryNotEmptyException, IOException,
+      InvalidPathException {
+    DeleteContext syncDeleteContext = DeleteContext.mergeFrom(
+            DeletePOptions.newBuilder()
+                .setRecursive(true)
+                .setAlluxioOnly(true)
+                .setUnchecked(true))
+        .skipNotPersisted(skipNonPersisted)
+        .setMetadataLoad(true);
+    Pair<Integer, Integer> deletedInodes = mFsMaster.deleteInternal(context.getRpcContext(),
+        lockedInodePath, syncDeleteContext, true);
+    if (deletedInodes.getFirst() == 0 && deletedInodes.getSecond() == 0) {
+      throw new FileDoesNotExistException(lockedInodePath + " does not exist.");
+    }
+    return deletedInodes;
+  }
+
+  private void updateInodeMetadata(
+      SyncProcessContext context, LockedInodePath lockedInodePath,
+      UfsStatus ufsStatus, Fingerprint fingerprint)
+      throws FileDoesNotExistException, AccessControlException, InvalidPathException {
+    // UpdateMetadata is used when a file or a directory only had metadata change.
+    // It works by calling SetAttributeInternal on the inodePath.
+    short mode = ufsStatus.getMode();
+    SetAttributePOptions.Builder builder = SetAttributePOptions.newBuilder()
+        .setMode(new Mode(mode).toProto());
+    if (!ufsStatus.getOwner().equals("")) {
+      builder.setOwner(ufsStatus.getOwner());
+    }
+    if (!ufsStatus.getGroup().equals("")) {
+      builder.setOwner(ufsStatus.getGroup());
+    }
+    SetAttributeContext ctx = SetAttributeContext.mergeFrom(builder)
+        .setUfsFingerprint(fingerprint.serialize())
+        .setMetadataLoad(true);
+    // Why previously clock is used?
```

Comment:
Is there anything you need to change for this? @elega

Reply:
no i removed this comment. i just do not understand where elsewhere uses Clock::mills() instead of CommonUtils.getCurrentMs(). Was that one more efficient?